File: 3683-find-the-lexicographically-largest-string-from-the-box-i/find-the-lexicographically-largest-string-from-the-box-i.cpp
class Solution {
public:
    string answerString(string word, int numFriends) {
        int n = word.size();

        int m = n - numFriends + 1;

        if(numFriends == 1)
        return word;

        string result;

        for(int i = 0; i < n; i++){
            result = max(result, word.substr(i, m));
        }
        return result;
    }
};

File: 3683-find-the-lexicographically-largest-string-from-the-box-i/find-the-lexicographically-largest-string-from-the-box-i.py
class Solution(object):
    def answerString(self, word, numFriends):
        """
        :type word: str
        :type numFriends: int
        :rtype: str
        """

        n = len(word)
        m = n - numFriends + 1

        if numFriends == 1: 
          return word

        return max(word[i : i + m] for i in range(n))
        